### ISSUE TITLE
Auto-creation of framework hooks

### ISSUE THREAD

**Issue by iamtrask** [OpenMined/PySyft#2620]
**Is your feature request related to a problem? Please describe.**
I find it frustrating that every time I import syft I have to create a hook
```
import syft as sy
import torch as th
hook = sy.TorchHook(th)
```
**Describe the solution you'd like**
I'd like to have an import option which creates the hook automatically. I.e.
```
import syft_torch as sy
th = sy.th
```
and then just start coding, with a "hook" variable created automatically and put on sy (sy.hook).

**Comment by jvmancuso** [comment]
There might be a concern here when we're using multiple frameworks in one script -- the global state of `syft.hook` might need to reference more than one hook.  This can be solved by assigning each one to a specific `syft.framework.hook`, i.e. `syft.torch.hook`, but then that's a lot of global state to maintain, so we'd probably want a solid way of dealing with that.  Right now it's a bit hand wavy how this is handled, but it's definitely been on my todo list.

**Comment by iamtrask** [comment]
Great point - although I think my assumption was that you'd only use these
if you knew which framework you were using (and you were only using one)

**Comment by jvmancuso** [comment]
Yeah, I've been leaning toward a combination here.  Currently, `syft.torch =TorchAttributes(torch)` is still called during `TorchHook.__init__`, but I'm assigning `syft.framework = syft.torch` as well.  `syft.framework` manages which framework & hook are currently being used, and when you switch between e.g. torch and tensorflow there would be some implicit switch of `syft.framework = syft.tensorflow`.  This allows generic/framework-agnostic code to call e.g. `syft.framework.hook` without needing to know which framework is currently in use. This switching needs to be a first-class mechanism though, and right now it's not, so that's what's been on my todo list.

**Comment by iamtrask** [comment]
Love it! Maybe close this PR when you merge?

**Comment by robert-wagner** [comment]
We actually tried this before when we only had pytorch (I might still have the branch somewhere) Creating the hook automatically causes so many problems when you are importing syft in other parts of the library as well as when you are just running it through an autoformatter/the test suite etc. We probably could hack around that but it would not be fun

**Comment by robert-wagner** [comment]
All for Jasons changes assuming they dont have similar problems

**Comment by iamtrask** [comment]
This is why I thought that it should be it's own package. syft itself
shouldn't create a hook, but syft_torch should (and shouldn't be imported
internally anywhere)

**Comment by jvmancuso** [comment]
Ah, yeah, so that is definitely a separate issue than managing multiple hooks.  Right now syft_torch etc. need to be imported inside `syft/frameworks/__init__.py`, and since `syft/__init__.py` imports that directory, it's roughly the same as doing it in syft.  I'd suggest we leave this issue blocked by the work I mentioned above, and resume conversation once that stuff lands.

**Comment by iamtrask** [comment]
No no - I'm not advocating for *anything* to call syft_torch (ever). I'm
saying we should create a separate PyPI package which literally just
imports syft, torch, and creates a hook for you.

**Comment by jvmancuso** [comment]
Oh, got confused because the pythonic name of PySyft-PyTorch will inevitably be `syft_torch`.
